src/solver/pooling/forward2d.cpp

```diff
@@ -37,7 +37,101 @@ namespace solver {
 
 namespace pooling {
 
-bool PoolingForward2d::IsApplicable(const ExecutionContext&,
+namespace {
+
+struct kernel_params
+{
+    int kernel_size_h;
+    int kernel_size_w;
+    int kernel_stride_h;
+    int kernel_stride_w;
+    int out_height;
+    int out_width;
+    int out_pix_tile0;
+    int out_pix_tile1;
+
+    kernel_params(const miopen::pooling::ProblemDescription& p)
+    {
+        const auto& pd  = p.GetPooling();
+        const auto& yd  = p.GetYDesc();
+        kernel_size_h   = pd.lens[0];
+        kernel_size_w   = pd.lens[1];
+        kernel_stride_h = pd.strides[0];
+        kernel_stride_w = pd.strides[1];
+        out_height      = yd.GetLengths()[2];
+        out_width       = yd.GetLengths()[3];
+        out_pix_tile0   = 1;
+        out_pix_tile1   = out_height <= 8    ? 1 //
+                          : out_height <= 32 ? 4 //
+                                             : 8;
+        if(out_height > 16 && out_height % 32 > 16)
+            out_pix_tile1 = std::min(16, std::max(1, prePow2(out_pix_tile1 * kernel_stride_h)));
+    }
+};
+
+std::size_t sizeof_kernel_FLOAT(const miopen::pooling::ProblemDescription& problem)
+{
+    const auto datatype = problem.GetXDesc().GetType();
+    return get_data_size(datatype);
+}
+
+std::size_t sizeof_kernel_FLOAT_ACCUM(const miopen::pooling::ProblemDescription& problem)
+{
+    const auto datatype = problem.GetXDesc().GetType();
+    if(datatype == miopenHalf)
+        return get_data_size(miopenFloat); // mixed precision
+    return get_data_size(datatype);
+}
+
+inline std::size_t RoundUpToMultiple(std::size_t v, std::size_t m)
+{
+    assert(m > 0);
+    return ((v + m - 1) / m) * m;
+}
+
+// Compute amount of private memory required for holding the arrays defined
+// in the "mloPoolingG" kernel:
+//
+// #define MLO_BOT_DATA_SZ0 \
```

Comment:
@atamazov :  there is another tidy issue
```
[2023-05-01T16:28:47.204Z] /home/jenkins/workspace/MLLibs_MIOpen_PR-2118/src/solver/pooling/forward2d.cpp:95:1: error: multi-line comment [-Werror=comment]

[2023-05-01T16:28:47.204Z]    95 | // #define MLO_BOT_DATA_SZ0 \

[2023-05-01T16:28:47.204Z]       | ^

[2023-05-01T16:28:47.204Z] /home/jenkins/workspace/MLLibs_MIOpen_PR-2118/src/solver/pooling/forward2d.cpp:97:1: error: multi-line comment [-Werror=comment]

[2023-05-01T16:28:47.204Z]    97 | // #define MLO_BOT_DATA_SZ1 \

[2023-05-01T16:28:47.204Z]       | ^
```

Reply:
And another issue:
```
cc1plus: error: unrecognized command line option ‘-Wno-option-ignored’ [-Werror]
```

Reply:
@junliume 
> ```
> cc1plus: error: unrecognized command line option ‘-Wno-option-ignored’ [-Werror]
> ```

This is from Jenkinsfile or from ./cmake/EnableCompilerWarnings.cmake, this PR has nothing to do with it.

> ```
> [2023-05-01T16:28:47.204Z] /home/jenkins/workspace/MLLibs_MIOpen_PR-2118/src/solver/pooling/forward2d.cpp:95:1: error: multi-line comment [-Werror=comment]
> ```

This will be fixed soon